src/server/view/DetectionPanel.java

```diff
@@ -151,7 +157,303 @@ public DetectionPanel() {
         PerformanceMetricsSpinner.setBounds(161, 229, 55, 25);
         this.add(PerformanceMetricsSpinner);
         
+        upperfaceSpinner.addChangeListener(new ChangeListener() {
+            @Override
+            public void stateChanged(ChangeEvent e) {
+                float upperfaceValue = (float)((double) upperfaceSpinner.getValue());
+                String upperfaceExp = (String)UpperfaceComboBox.getSelectedItem();
+                if(upperfaceExp=="Raise Brow")
+                {
+                    ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setRaiseBrow(upperfaceValue);
+                    ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setFurrowBrow(0);
+                }
+                if(upperfaceExp=="Furrow Brow")
+                {
+                    ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setFurrowBrow(upperfaceValue);
+                    ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setRaiseBrow(0);
+                }
+
+            }
+        });
+         
+        UpperfaceComboBox.addActionListener(
+                 new ActionListener(){
+                 public void actionPerformed(ActionEvent e){
+                 float upperfaceValue = (float)((double) upperfaceSpinner.getValue());
+                 String upperfaceExp = (String)UpperfaceComboBox.getSelectedItem();
+                 if(upperfaceExp=="Raise Brow")
+                 {
+                	 ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setRaiseBrow(upperfaceValue);
+                     ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setFurrowBrow(0);
+                 }
+                 if(upperfaceExp=="Furrow Brow")
+                 {
+                	 ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setFurrowBrow(upperfaceValue);
+                     ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setRaiseBrow(0);
+                 }
+            }
+         });
+   
+        lowerfaceSpinner.addChangeListener(new ChangeListener() {
+            @Override
+            public void stateChanged(ChangeEvent e) {
+                float lowerfaceValue = (float)((double) lowerfaceSpinner.getValue());
+                String lowerfaceExp = (String)LowerfaceComboBox.getSelectedItem();
+                if(lowerfaceExp=="Smile")
+                {
+                    ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setSmile(lowerfaceValue);
+                    ServerDataSingleton.getInstance().getFaceData().getExpressiveData().setClench(0);
```

Comment:
Instead of setting individual values every time.. We have a generic function which sets all the value to zero. And then we set the required field.

Reply:
This would reduce the implementation significantly.